internal/proxyservice/tcp/pg/backend.go

```diff
@@ -2,51 +2,46 @@ package pg
 
 import (
 	"fmt"
-	"log"
 	"net"
 	"reflect"
 
-	"github.com/cyberark/secretless-broker/internal/handlers/pg/protocol"
 	"github.com/cyberark/secretless-broker/internal/handlers/ssl"
-	"github.com/cyberark/secretless-broker/internal/util"
+	"github.com/cyberark/secretless-broker/internal/proxyservice/tcp/pg/protocol"
 )
 
 // ConfigureBackend resolves the backend connection settings and credentials and sets the
 // connectionDetails field.
-func (h *Handler) ConfigureBackend() error {
+func (s *SingleUseConnector) ConfigureBackend(credentials map[string][]byte) error {
```

Comment:
(make sure to read the security comment below, as it's related to this one...)

I know this is a carryover from the old code, but this name isn't really accurate or descriptive.  And since we're rewriting, we should improve the old mistakes. 

I have two thoughts:

1. `BackendConfig` -> `ConnectionDetails` (name we use in mysql, which is good)
2. _drumroll please...._ 

DELETE this method entirely.  If you think about what it's doing, it's quite silly.  Instead, we should move the relevant content that's in this method into `NewBackendConfig` / `NewConnectionDetails`.  Then update this part of `Connect`: 
```go
	if err := s.ConfigureBackend(credentialValuesByID); err != nil {
		s.abort(err)
		return nil, err
	}
```
to something like:
```go
connDetails, err := NewConnectionDetails(credentialValuesByID); err != nil {
 		s.abort(err)
		return nil, err
}
s.ConnectionDetails = connDetails
```